lib/src/Pos/Diffusion/Full/Block.hs

```diff
@@ -593,7 +592,7 @@ blockListeners
     -> ProtocolConstants
     -> Word
     -> OQ.OutboundQ pack NodeId Bucket
-    -> Timer -- ^ Keepalive timer
+    -> (NodeId -> IO ()) -- ^ Keepalive timer
```

Comment:
Why not add a newtype wrapper `ResetNodeTimer` for this.  It will convey the meaning of this `IO` action.

Reply:
We completed cardano-sl release/2.0.0. I  will check whether we need to have this as a hotfix.